server/principal.go

```diff
@@ -22,7 +22,15 @@ func (s *Server) registerPrincipalRoutes(g *echo.Group) {
 		}
 
 		principalCreate.CreatorID = c.Get(getPrincipalIDContextKey()).(int)
-		principalCreate.Type = api.EndUser
+
+		if principalCreate.Type == api.ServiceAccount {
+			pwd, err := common.RandomString(20)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to generate access key for service account.").SetInternal(err)
+			}
+			principalCreate.Password = fmt.Sprintf("bb.service_account.%s", pwd)
```

Comment:
I would choose bbs_randomstring. To make it look more cryptic (because it's a secret) , bb.service_account looks too structured and may not convey the secrecy.

bb stands for bytebase, s stands for service account or server-to-server token

https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/